leo/cli/commands/add.rs

```diff
@@ -53,32 +56,61 @@ impl Command for Add {
         let manifest: Manifest = serde_json::from_str(&program_data)
             .map_err(|err| PackageError::failed_to_deserialize_manifest_file(path.to_str().unwrap(), err))?;
 
-        // Allow both `credits.aleo` and `credits` syntax
-        let name = match self.name {
-            name if name.ends_with(".aleo") => name,
-            name => format!("{}.aleo", name),
+        // Make sure the program name is valid.
+        // Allow both `credits.aleo` and `credits` syntax.
+        let name: String = match &self.name {
+            name if name.ends_with(".aleo") && is_valid_program_name(&name[0..self.name.len() - 5]) => name.clone(),
+            name if is_valid_program_name(name) => format!("{name}.aleo"),
+            name => return Err(PackageError::invalid_file_name_dependency(name).into()),
         };
 
         // Add dependency section to manifest if it doesn't exist
-        let mut dependencies = match manifest.dependencies() {
-            Some(ref dependencies) => dependencies
+        let mut dependencies = match (self.clear, manifest.dependencies()) {
+            (false, Some(ref dependencies)) => dependencies
                 .iter()
                 .filter_map(|dependency| {
+                    // Overwrite old dependencies of the same name.
                     if dependency.name() == &name {
-                        println!("{} already exists as a dependency. Overwriting.", name);
+                        let msg = match (dependency.path(), dependency.network()) {
+                            (Some(local_path), _) => {
+                                format!("local dependency at path `{}`", local_path.to_str().unwrap().replace('\"', ""))
+                            }
+                            (_, Some(network)) => {
+                                format!("network dependency from `{}`", network)
+                            }
+                            _ => "git dependency".to_string(),
+                        };
+                        tracing::warn!("⚠️  Program `{name}` already exists as a {msg}. Overwriting.");
```

Comment:
Should we error and tell the user to clear the dependency?

Reply:
I worry this may make the dependency process cumbersome

Reply:
One thing I noticed is that when you add a dependency in cargo that already exists (e.g. a newer version) then it replaces it automatically